drivers/soundwire/intel.c

```diff
@@ -1249,10 +1254,12 @@ static struct sdw_master_ops sdw_intel_ops = {
 
 static int intel_init(struct sdw_intel *sdw)
 {
+	mutex_lock(sdw->link_res->shim_lock);
+
 	/* Initialize shim and controller */
 	intel_link_power_up(sdw);
 
-	intel_shim_init(sdw);
+	mutex_unlock(sdw->link_res->shim_lock);
```

Comment:
any specific reason why both have to be locked together? If anything this introduces an asymmetry with `intel_link_power_down()` which takes the lock internally

Reply:
well, I could remove the wrapper now and use power_up directly with an internal lock taken,

I haven't had time to look into this PR, it's on the backburner for now.